### PR TITLE
Forbid test sources to use System.out.println and Throwable.printStackTrace

### PR DESCRIPTION
**Pull request by ywelsch**
Test sources have been exempt from the bundled forbiddenApi signatures `jdk-system-out` that check among others usages of `System.out.println` and `Throwable.printStackTrace`. With the removal of benchmarks in #15356, there is no need to relax this check for tests anymore.

This PR:
- enables `jdk-system-out` checks on all sources
- removes the few remaining usages of println and printStackTrace
- removes two benchmark classes (probably missed in #15356)

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/http/netty/NettyHttpServerPipeliningTests.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ public void messageReceived(final ChannelHandlerContext ctx, final MessageEvent
 
         @Override
         public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) {
-            e.getCause().printStackTrace();
+            logger.info("Caught exception", e.getCause());
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/http/netty/NettyHttpServerPipeliningTests.java]: Just log `e`?

File: test/logger-usage/src/main/java/org/elasticsearch/test/loggerusage/ESLoggerUsageChecker.java [hunk 2]
```diff
@@ -59,6 +57,7 @@
     public static final List<String> LOGGER_METHODS = Arrays.asList("trace", "debug", "info", "warn", "error");
     public static final String IGNORE_CHECKS_ANNOTATION = "org.elasticsearch.common.SuppressLoggerChecks";
 
+    @SuppressForbidden(reason = "command line tool")
```
Comment (s1monw) [inline test/logger-usage/src/main/java/org/elasticsearch/test/loggerusage/ESLoggerUsageChecker.java]: can you use the `@ SuppressForbidden` from lucene here? I think the test framework has dependencies on lucene no?

**Comment by s1monw** [comment]
left one minor comment - LGTM otherwise, no need to re-review

File: test/logger-usage/src/main/java/org/elasticsearch/test/loggerusage/ESLoggerUsageChecker.java (hunk 2 above)
Comment (ywelsch) [inline test/logger-usage/src/main/java/org/elasticsearch/test/loggerusage/ESLoggerUsageChecker.java]: The logger usage checker is a standalone tool. Its main sources have no dependencies besides ASM framework (its test sources reuse our test framework however). Just one additional "SuppressForbidden" beside the other 5 in the classpath ;-)

**Comment by jasontedor** [comment]
LGTM.

**Pull request merged**